File: RootsOfFunctions/bisection.jl
function bisection(a, b, ntol)
	f(z) = z^3 - z + 1
	x = 0.0
	for n = 1:ntol
		x = a + (b - a)/2
		if f(x) * f(b) < 0
			a = x
		else
			b = x
		end
	end
	return x
end

println(bisection(-5,0, 10))
		

File: RootsOfFunctions/bisection.py
def f(x):
    return x ** 3 - x + 1

def bisection(a, b, ntol):
    for n in range(0,ntol):
        x = a + (b - a) / 2
        if f(x) * f(b) < 0:
            a = x
        else:
            b = x
    return x

print(bisection(-5, 0, 10))
